### ISSUE TITLE
Cannot getSize() when pulling data over HTTP / HTTPS

### ISSUE THREAD

**Issue by afrosimon** [DataBiosphere/toil#2986]
I stumbled upon an exception when toil is trying to get the size of a file over HTTP, based on the Content-Length header. Thing is, this header is not always present, for instance when a file is transferred over Transfer-Encoding: chunked; (as per HTTP 1.1 standard, when using this transfer method, the Content-Length should NOT be there).

A dumb hotfix that worked for me was to provide a default value of 0 when that header is absent, but I'm concerned this may have side effects somewhere else. I can see the getSize method for the jobStores is used seldomly, mostly the size seems to be used in the FileID class when packing / unpacking (I do not know enough about toil currently to extrapolate any further).

So basically I'd like to create a PR for this eventually, anyone could chime in and provide some more context / some pointers as to what other use case I should test for, etc..

You can see my fix there : https://github.com/afrosimon/toil/commit/98f33dd10cc6123d9ae80553847864343f5d884d

Here's a stacktrace for the original exception : 
```
File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/wdl/wdl_functions.py", line 166, in process_single_infile
    filepath = fileStore.importFile(f)
  File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/common.py", line 979, in importFile
    return self._jobStore.importFile(srcUrl, sharedFileName=sharedFileName)
  File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/jobStores/abstractJobStore.py", line 304, in importFile
    return self._importFile(otherCls, srcUrl, sharedFileName=sharedFileName, hardlink=hardlink)
  File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/jobStores/fileJobStore.py", line 291, in _importFile
    sharedFileName=sharedFileName)
  File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/jobStores/abstractJobStore.py", line 326, in _importFile
    return FileID(jobStoreFileID, otherCls.getSize(url))
  File "/SOME_PROJECT/env/lib/python3.7/site-packages/toil/jobStores/abstractJobStore.py", line 1067, in getSize
    return int(readable.info().get('content-length'))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-516)
┆Issue Number: TOIL-516

**Comment by adamnovak** [comment]
I think at the point where we are calling `getSize` we have already copied the whole file into the job store.

We should probably ask the job store about the size of the file, since we already implemented size polling there, instead of trying to get it form the URL we imported.

I don;t think we want to report a file size of 0; that will confuse caching. If the file size turns out to be really unavailable somehow, we could return just a string and not a FileID, and places that need the size will poll for it themselves. But that's just postponing the actual measurement.
